apps/src/lab2/views/components/Instructions/ResourcePanel/index.tsx

```diff
@@ -191,8 +202,11 @@ const ResourcePanel: React.FC<ResourcePanelProps> = ({
     // a student's project (in which case they can view old versions, but not restore them).
     // We never show the version history tab in widget view, as widget view is always read-only
     // and therefore can never have version history.
+    // Note: We ignore the temporary read-only state caused by running/validating code when a user clicks on 'Run' or 'Validate' buttons.
+    const isPermanentlyReadOnly = isReadOnly && !isRunning && !isValidating;
```

Comment:
This was a particularly sticky bug that took me a while to figure out. On a standalone `pythonlab` project, there is only one tab (version history). When the user runs the program, the project is set to read-only temporarily so that the version history tab is no longer existent and this caused some unexpected behavior (resource panel collapsed since there were now 0 tabs and when expanded, no content was rendered).
Thus, we distinguish between 'permanent' and 'temporary' read-only states.

Reply:
Ahh..yeah - I'll disable when it's in this temp read-only state. Thanks!

Reply:
After latest update:

https://github.com/user-attachments/assets/69ecf7b9-a860-4e96-b81b-e82580d80589

Reply:
This is still a little buggy...will investigate.